source/appModules/soffice.py

```diff
@@ -340,3 +342,30 @@ def event_NVDAObject_init(self, obj):
 			# This is a word processor document.
 			obj.description = None
 			obj.treeInterceptorClass = SymphonyDocument
+
+	def searchStatusBar(self, obj: NVDAObject, max_depth: int = 5) -> Optional[NVDAObject]:
+		"""Searches for and returns the status bar object
+		if either the object itself or one of its recursive children
+		(up to the given depth) has the corresponding role."""
+		if obj.role == controlTypes.Role.STATUSBAR:
+			return obj
+		if max_depth < 1 or obj.role not in {controlTypes.Role.ROOTPANE, controlTypes.Role.WINDOW}:
+			return None
+		for child in obj.children:
+			status_bar = self.searchStatusBar(child, max_depth - 1)
+			if status_bar:
+				return status_bar
+		return None
+
+	def _get_statusBar(self):

```

Comment:
Added a return type for this

```suggestion
	def _get_statusBar(self) -> Optional[NVDAObject]:
```

Reply:
Done